it/controllers/agent/incomeSources/cease/CheckCeaseIncomeSourceDetailsControllerISpec.scala

```diff
@@ -0,0 +1,214 @@
+  val requestUK: UpdateIncomeSourceRequestModel = UpdateIncomeSourceRequestModel(
+    nino = testNino,
+    incomeSourceId = ukProperty.incomeSourceId,
+    cessation = Some(Cessation(true, Some(LocalDate.parse(cessationDate))))
+  )
+
+  val sessionCeaseForeignPropertyEndDate = Map(forms.utils.SessionKeys.ceaseForeignPropertyEndDate -> cessationDate)
+  val showCheckCeaseForeignPropertyDetailsControllerUrl = controllers.incomeSources.cease.routes.CeaseCheckIncomeSourceDetailsController.showAgent(ForeignProperty).url
+  val submitCheckCeaseForeignPropertyDetailsControllerUrl = controllers.incomeSources.cease.routes.CeaseCheckIncomeSourceDetailsController.showAgent(ForeignProperty).url
+
+  val formActionFP = controllers.incomeSources.cease.routes.CeaseCheckIncomeSourceDetailsController.submitAgent(ForeignProperty).url
+  val pageTitleMsgKeyFP = messagesAPI("incomeSources.ceaseForeignProperty.checkDetails.heading")
+  val redirectUriFP = controllers.incomeSources.cease.routes.IncomeSourceCeasedObligationsController.showAgent(ForeignProperty).url
+  val requestFP: UpdateIncomeSourceRequestModel = UpdateIncomeSourceRequestModel(
+    nino = testNino,
+    incomeSourceId = foreignProperty.incomeSourceId,
+    cessation = Some(Cessation(true, Some(LocalDate.parse(cessationDate))))
+  )
+
+  s"calling GET ${showCheckCeaseBusinessDetailsControllerUrl}" should {
+    "render the Cease Business Page" when {
+      "User is authorised" in {
+        enable(IncomeSources)
+        stubAuthorisedAgentUser(authorised = true)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(
+          status = OK,
+          response = businessAndPropertyResponse
+        )
+
+        val res = IncomeTaxViewChangeFrontend.getCheckCeaseBusinessDetails(sessionCeaseBusinessDetails ++ clientDetailsWithConfirmation)
+        verifyIncomeSourceDetailsCall(testMtditid)
+
+        res should have(
+          httpStatus(OK),
+          pageTitleAgent(pageTitleMsgKey),
+          elementTextBySelectorList(".govuk-summary-list__key", "dt:nth-of-type(1)")(messagesAPI("incomeSources.ceaseBusiness.checkDetails.dateStopped")),
+          elementTextBySelectorList(".govuk-summary-list__value", "dd:nth-of-type(1)")(businessEndShortLongDate),
+          elementTextByID("change-link")(changeLink),
+
+          elementTextBySelector("dl:nth-of-type(1) div:nth-of-type(2) dt:nth-of-type(1)")(messagesAPI("incomeSources.ceaseBusiness.checkDetails.businessName")),
+          elementTextBySelector("dl:nth-of-type(1) div:nth-of-type(2) dd:nth-of-type(1)")(testBusinessName),
+          elementAttributeBySelector("form", "action")(formActionSE)
+        )
+      }
+    }
+  }
+
+  s"calling POST ${showCheckCeaseBusinessDetailsControllerUrl}" should {
+    s"redirect to $redirectUriSE" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with multiple business and property")
```

Comment:
I think you're only stubbing a single business (line 102)

Reply:
addressed all test comments